### ISSUE TITLE
GLM_FORCE_ALIGNED breaks gtc/random ballRand

### ISSUE THREAD

**Issue by chrisvarns** [g-truc/glm#807]
I believe this is related to https://github.com/g-truc/glm/issues/762

The following code fails to compile:
```
#define GLM_FORCE_ALIGNED
...
vec3 test = vec3(0.0f) + ballRand(1.0f);
```

with the error: 
> Invalid operands to binary expression ('const glm::vec3' (aka 'const vec<3, float, highp>') and 'vec<3, float, defaultp>')

I believe the issue is that the lhs is vec3, which is a typedef of highp_vec3, and the qualifier is packed_highp, but ballRand is returning one with aligned_highp and there is no overload of operator+ that can handle this?

**Comment by chrisvarns** [comment]
If glm::vec3 does not react to GLM_FORCE_ALIGNED, I am probably being dense but what is the intended usage when we want to guarantee the SIMD paths are taken?

**Comment by Groovounet** [comment]
I believe this was a bug in GLM 0.9.9.0 which is resolved in master branch for GLM 0.9.9.1.

GLM_FORCE_ALIGNED still works but was renamed into GLM_FORCE_DEFAULT_ALIGNED_GENTYPES.

Thanks for contributing!
